### ISSUE TITLE
[Bug] Hans with localized LANG-0.DAT RUS

### ISSUE THREAD

**Issue by BoresXP** [CorsixTH/CorsixTH#2066]
#### Steps to Reproduce
1. Use data file LANG-0.DAT of the edition localized to Russian (official).
2. Programs hangs after showing CD with doctors image.

#### Expected Behavior
Starts as usual

#### System Information
**CorsixTH Version:** 
0.65.1

**Operating System:** 
Windows 10 x64 21H1 19043.1237

**Theme Hospital Version:*
CD image of Russian localization.
Can be downloaded here: https://www.old-games.ru/game/download/get.php?fileid=26584&modal=1

#### Gamelog.txt
Gamelog not created for this issue

**Comment by BoresXP** [comment]
File that caused this freeze
[LANG-0.zip](https://github.com/CorsixTH/CorsixTH/files/7279909/LANG-0.zip)
.

**Comment by BoresXP** [comment]
> I  am suspecting more than just the the language and sound files were changed on this image

I've made a simple check: take original ENG image that runs in CorsixTX without issues. Copy only LANG-0 from RUS CD image over it. And try to run. CorsixTH hangs on start.
So I'm absolutely sure that LANG-0 causes CorsixTX to hang. Not other files.

**Comment by TheCycoONE** [comment]
How did you obtain this version in the first place. We officially don't support 'abandonware' downloads.

I'll take a look at the lang-0

**Comment by lewri** [comment]
I tried to not interfere with english by having changing it to Lang-6, then pointing the russian.lua to inherit original strings with 6 defined.

**Comment by TheCycoONE** [comment]
It's at least based off CP866, I got several accurate looking translations and some nonsense but my Russian isn't good enough to say whether that's because the encoding is off or because of separators in the language file. @BoresXP are you able to verify the encoding? Try opening as CP866 and see if it looks right.

**Comment by BoresXP** [comment]
How can I help you make this bug fixed?

**Comment by lewri** [comment]
I believe we need to know the exact encoding type of the language. Any idea what the characters are meant to be where it is garbled? 
I'm interested too how the Russian "sounds" work. Do you mean someone redid the receptionist announcements in Russian?

**Comment by BoresXP** [comment]
It's mostly CP866 with some exception. I can make a mapping (chat - code) for exceptions if it can help.

Sound announcements if Russian works fine by just coping them over original ones. I'm playing right with such mix (sounds from officially localized version and texts from Corsix).

**Comment by lewri** [comment]
Yes a mapping would likely help out getting closer and save cyco a bit of work. Not sure how it'll be implemented into the code though (I'm thinking it'll be some kind of rule where it looks for a Russian language file (e.g. call it lang-6) and if it can't find it then fall back to the corsix strings... Maybe.)

**Comment by TheCycoONE** [comment]
It would have come with a different sprite sheet too. I'll need that documented. This is all pretty interesting.

**Comment by BoresXP** [comment]
So, I made a substitution table:
`Р (0x90) = 0xB0
С (0x91) = 0xB1
Т (0x92) = 0xB2
У (0x93) = 0xB3
Ф (0x94) = 0xB4
Х (0x95) = 0xB5
Ц (0x96) = 0xB6
Ч (0x97) = 0xB7
Ш (0x98) = 0xB8
Щ (0x99) = 0xB9
Ъ (0x9A) = 0xBA
Ы (0x9B) = 0xBB
Ь (0x9C) = 0xBC
Э (0x9D) = 0xBD
Ю (0x9E) = 0xBE
Я (0x9F) = 0xBF`
If you look at CP866 character table (https://en.wikipedia.org/wiki/Code_page_866) you will find that it is just second row (9x) changed to fourth (Bx). 
But not everywhere! In names all characters stay at correct places. 
Looks like is is some workaround for construction blocks.

**Comment by TheCycoONE** [comment]
How tricky - they might have used different code pages for different fonts then?

**Comment by BoresXP** [comment]
> How tricky - they might have used different code pages for different fonts then?

Looks like so

**Comment by lewri** [comment]
Do we wish to keep pursuing this one @TheCycoONE ?

**Comment by ARGAMX** [comment]
We may not support these third party `LANG-0.DAT` files.

But,
It seems we should display an error about this issue in some more explicit form and not just freeze the game.

In this case it looks like game stuck because of "Unicode font not found, backup is unavailable":

But in fact the problem here is in `LANG-0.DAT`. But it is not obvious at all.

![Image](https://github.com/user-attachments/assets/d9fa4605-070c-4d44-85da-da2a951e9b42)

**Comment by TheCycoONE** [comment]
I think the actual encoding is https://en.wikipedia.org/wiki/MIK_(character_set) - this produces better strings like:

Медсестра Врач Рабочий Регистратор Хирург
